internal/message/upload.go

```diff
@@ -262,3 +263,40 @@ func MoveMessage(srcSection Section, targetSection Section, msgID string) error
 
 	return fs.Rename(p1, p2)
 }
+
+// StoreLocalMessage will store a message locally
+func StoreLocalMessage(header *Header, catalog io.Reader, blocks map[string]*io.Reader, attachments map[string]*io.Reader) (string, error) {
```

Comment:
Could we use an envelope here instead of sending all these arguments (not sure if that's possible though)

Reply:
definitely! ouch!